src/display/api.js

```diff
@@ -1577,30 +1579,97 @@ var WorkerTransport = (function WorkerTransportClosure() {
       var pdfDataRangeTransport = this.pdfDataRangeTransport;
       if (pdfDataRangeTransport) {
         pdfDataRangeTransport.addRangeListener(function(begin, chunk) {
-          messageHandler.send('OnDataRange', {
-            begin,
-            chunk,
-          });
+          this._networkStream.onReceivedData({ begin, chunk, });
         });
 
         pdfDataRangeTransport.addProgressListener(function(loaded) {
-          messageHandler.send('OnDataProgress', {
-            loaded,
-          });
+          this._networkStream.onDataProgress({ loaded, });
         });
 
         pdfDataRangeTransport.addProgressiveReadListener(function(chunk) {
-          messageHandler.send('OnDataRange', {
-            chunk,
-          });
+          this._networkStream.onReceivedData({ chunk, });
         });
 
-        messageHandler.on('RequestDataRange',
-          function transportDataRange(data) {
+        this.networkStreamCapability.promise.then(() => {
+          this._networkStream.requestDataRangeCapability.promise.
+              then(function(data) {
             pdfDataRangeTransport.requestDataRange(data.begin, data.end);
-          }, this);
+          });
+        });
       }
 
+      messageHandler.on('GetReader', function(data, sink) {
+        data.disableRange = true;
+        data.disableStream = true;
+        if (data.chunkedViewerLoading) {
+          this._networkStream = new PDFWorkerStream(data.source);
+        } else {
+          this._networkStream = new PDFNetworkStream(data);
+        }
```

Comment:
move this code to the WorkerTransport constructor and remove this.networkStreamCapability promise